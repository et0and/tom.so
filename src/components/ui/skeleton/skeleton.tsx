import { cn } from "@/lib/utils";

function Skeleton({
  className,
  ...props
}: React.HTMLAttributes<HTMLDivElement>) {
  return (
    <div
      className={cn(
        "animate-pulse rounded-md bg-gradient-to-r from-gray-200 dark:bg-gray-200",
        className,
      )}
      {...props}
    />
  );
}

export { Skeleton };
