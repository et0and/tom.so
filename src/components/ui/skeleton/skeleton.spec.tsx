import React from "react";
import { render } from "@testing-library/react";
import { Skeleton } from "./skeleton";
import "@testing-library/jest-dom";

describe("<Skeleton/>", () => {
  it("renders correctly", () => {
    const { container } = render(<Skeleton />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("applies custom className", () => {
    const { container } = render(<Skeleton className="custom-class" />);
    expect(container.firstChild).toHaveClass("custom-class");
    expect(container.firstChild).toMatchSnapshot();
  });

  it("passes through additional props", () => {
    const { container } = render(<Skeleton data-testid="skeleton-test" />);
    expect(container.firstChild).toHaveAttribute(
      "data-testid",
      "skeleton-test",
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
