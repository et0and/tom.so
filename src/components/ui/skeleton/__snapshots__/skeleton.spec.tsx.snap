// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton/> applies custom className 1`] = `
<div
  class="animate-pulse rounded-md bg-gradient-to-r from-gray-200 dark:bg-gray-200 custom-class"
/>
`;

exports[`<Skeleton/> passes through additional props 1`] = `
<div
  class="animate-pulse rounded-md bg-gradient-to-r from-gray-200 dark:bg-gray-200"
  data-testid="skeleton-test"
/>
`;

exports[`<Skeleton/> renders correctly 1`] = `
<div
  class="animate-pulse rounded-md bg-gradient-to-r from-gray-200 dark:bg-gray-200"
/>
`;
